src/login/logind-session.c

```diff
@@ -664,12 +664,12 @@ static int session_start_scope(Session *s, sd_bus_message *properties, sd_bus_er
                                 description,
                                 /* These two have StopWhenUnneeded= set, hence add a dep towards them */
                                 STRV_MAKE(s->user->runtime_dir_service,
-                                          s->user->service),
+                                          s->class != SESSION_BACKGROUND ? s->user->service : NULL),
                                 /* And order us after some more */
                                 STRV_MAKE("systemd-logind.service",
                                           "systemd-user-sessions.service",
                                           s->user->runtime_dir_service,
-                                          s->user->service),
+                                          s->class != SESSION_BACKGROUND ? s->user->service : NULL),
```

Comment:
this one looks wrong.

in systemd ordering deps and requirement deps are unconditional. It might make sense to drop the requiement deps for some tasks, but i think the ordering should always be kept.

i.e. if something else pulls in user@.service anyway (for example: lingering is enabled for the user), then we should honour the order. And if it is not pulled in by anything the configured order has no effect. Hence we should just leave this in.

Reply:
Hmm, it makes sense to keep ordering if you don't strictly require other service but do use it in case it is pulled-in by some else. I don't want background instances to interact with user@.service in any way so I don't see why ordering would be beneficial.